Reparando bugs encontrados en las pruebas

josegomezr opened the pull request:
Con estos cambios logramos 99% de éxito en las pruebas. Hay 1 sola fallando referente a las validaciones de perfil.

Ya se han reparado todos los bugs encontrados (las marcas @todo) y mejorado el coverage, está más cerca del 100% en algunos archivos. Aunque aún falta refractor para lograr el 100%.

josegomezr committed:
@wip Auth tests

Employer tests listo para `list`, `show`, `update`

josegomezr committed:
@wip Auth tests

Employer tests listo para `list`, `show`, `update`

josegomezr committed:
@wip Auth tests

Pruebas al perfil

Nota: Hay 2 pruebas marcadas como fallos esperados:
- Perfil no existente
- Perfil con poca data

Hay una 3era recomendación (está comentada): si el perfil
es de employer, no deberia salir la clave `employee` o
viceversa, asi se expone menos información del endpoint.

josegomezr committed:
@wip Auth tests

Pruebas al perfil

Nota: Hay 2 pruebas marcadas como fallos esperados:
- Perfil no existente
- Perfil con poca data

Hay una 3era recomendación (está comentada): si el perfil
es de employer, no deberia salir la clave `employee` o
viceversa, asi se expone menos información del endpoint.

josegomezr committed:
@wip Test Auth Views

partial JobCore Invite tests

josegomezr committed:
@wip Test Auth Views

partial JobCore Invite tests

josegomezr committed:
@wip Rating Tests + unmarked all expectedFailure tests

What a day to run tests!

josegomezr committed:
@wip Rating Tests + unmarked all expectedFailure tests

What a day to run tests!

josegomezr committed:
Merge branch '2nd_phase'

josegomezr committed:
Bugfixing en login

josegomezr committed:
bugfixing en password reset y registration.

josegomezr committed:
Forzando ValidateEmailView.get() a sólo aceptar validar el email UNA SOLA VEZ.

josegomezr committed:
100% de pruebas pasadas! ahora podemos hacer más!

josegomezr committed:
Mezclando todo!

josegomezr committed:
mandando a saltar las pruebas legacy

josegomezr committed:
eliminando @todos

alacret commented on api/actions/auth_actions.py:
```diff
@@ -1,14 +1,17 @@
     for invite in jc_invites:
         if invite.shift.starting_at > timezone.now():
-            invite = ShiftInvite(sender=invite.sender, shift=invite.shift, employee=employee)
+            invite = ShiftInvite(
```
reemplazar con bulk create

josegomezr replied:
esto es legacy code...

alacret replied:
Lo entiendo, reparemoslo

alacret commented on api/serializers/auth_serializer.py:
```diff
@@ -24,63 +33,58 @@ class Meta:
-            msg = 'Account with this email does not exists'
+
+        try:
+            user_obj = User.objects.filter(lookup)[:1].get()
```
Por que de esta manera y no con un .first? o un `get` directo?

josegomezr replied:
Ah porque `[:1]` manda a escribir un `LIMIT 1` en la consulta generada. Para que `.get` no lanze un `MultipleObjectReturned` y siempre sea `DoesNotExist`

alacret replied:
reemplazalo por un first mejor, o por un get y revisas ambas excepciones

josegomezr replied:
again, legacy code...

alacret replied:
Lo entiendo, reparemoslo

alacret commented on api/serializers/auth_serializer.py:
```diff
@@ -24,63 +33,58 @@ class Meta:
+        device_id = attrs.get("registration_id")
+
+        if device_id is not None:
+            FCMDevice.objects.filter(registration_id=device_id).delete()
```
Haz un update o usa una transaccion

josegomezr replied:
Npi, esto es copypasteado de lo que estaba, exactamente que quieres decir con una TX? que el FCMDevice se "swap"-ee ? intercambio uno por otro pues?

alacret replied:
si, exacto, con el UPDATE garantizas que se mantenga consistente en cualquier fallo. Lo mismo lo haces con la transaccion. Pero creo que aqui el UPDATE sirve

alacret commented on api/serializers/auth_serializer.py:
```diff
@@ -90,89 +94,98 @@ class Meta:
         user.set_password(validated_data['password'])
         user.save()
-        
+
```
no hagas estos try multilineas, el try debe ser para una sentencia nada mas y recuperar

josegomezr replied:
again, legacy code...

alacret replied:
a este marcale con un todo, y luego lo hacemos

alacret commented on api/serializers/auth_serializer.py:
```diff
@@ -90,89 +94,98 @@ class Meta:
+
         try:
-                
+
```
considera abstraer la logica de la creacion de un empledao y un employer en 2 funciones distintas

alacret commented on api/serializers/rating_serializer.py:
```diff
@@ -91,8 +91,16 @@ def validate(self, data):
-                
-        
+
+            try:
```
confirma con @alesanchezr que esto es asi

josegomezr committed:
Linting fixes + Bugfixes

- Bulkcreate en vez de for-loop con .save() en auth_actions.
- Mejoras en el lookup del login
- Update en vez de Delete+Insert para FCM Device
- Eliminado un try/except block innecesario
- Mejoras en el algoritmo de puntuación (rating). Ahora usa SQL AVG/COUNT

josegomezr committed:
fixup en FCMDevice + respaldo en prueba

josegomezr committed:
making linter happy
